src/behavior.scm

```diff
@@ -758,38 +838,92 @@
         ))
 		(TrueLink)
 	))
+;;
+(define (pred-2-schema pnode-str) 
+	(DefineLink 
+		(DefinedSchemaNode pnode-str)
+		(ExecutionOutputLink (GroundedSchemaNode "scm: cog-evaluate!")
+			(ListLink (DefinedPredicateNode pnode-str))
+)))
+;;
+(pred-2-schema "Interaction requested action")
+(pred-2-schema "New arrival sequence action")
+(pred-2-schema "Someone left action")
+(pred-2-schema "Interact with people action")
+(pred-2-schema "Nothing is happening action")
+(pred-2-schema "Speech started? action")
+(pred-2-schema "Speech ongoing? action")
+(pred-2-schema "Speech ended? action")
+(pred-2-schema "Listening started? action")
+(pred-2-schema "Listening? action")
+(pred-2-schema "Listening ended? action")
+(pred-2-schema "Keep alive")
+;;
+;(DefineLink (DefinedPredicateNode "do-noop") (True))
+;(pred-2-schema "do-noop")
+
+(define demand-satisfied (True))
+(define speech-demand-satisfied (True))
+(define face-demand (psi-demand "face interaction" 1))
+(define speech-demand (psi-demand "speech interaction" 1))
+(define run-demand (psi-demand "run demand" 1))
+
+;(psi-rule (list (DefinedPredicate "Interaction requested"))(DefinedSchemaNode "Interaction requested action") demand-satisfied (stv 1 1) face-demand)
+;(psi-rule (list (DefinedPredicate "New arrival sequence")) (DefinedSchemaNode "New arrival sequence action") demand-satisfied (stv 1 1) face-demand)
+;(psi-rule (list (DefinedPredicate "Someone left")) (DefinedSchemaNode "Someone left action") demand-satisfied (stv 1 1) face-demand)
+;(psi-rule (list (DefinedPredicate "Interact with people")) (DefinedSchemaNode "Interact with people action") demand-satisfied (stv 1 1) face-demand)
+;(psi-rule (list (DefinedPredicate "Nothing is happening")) (DefinedSchemaNode "Nothing is happening action") demand-satisfied (stv 1 1) face-demand)
+;(psi-rule (list (DefinedPredicate "Speech started?")) (DefinedSchemaNode "Speech started? action") speech-demand-satisfied (stv 1 1) speech-demand)
+;(psi-rule (list (DefinedPredicate "Speech ongoing?")) (DefinedSchemaNode "Speech ongoing? action") speech-demand-satisfied (stv 1 1) speech-demand)
+;(psi-rule (list (DefinedPredicate "Speech ended?")) (DefinedSchemaNode "Speech ended? action") speech-demand-satisfied (stv 1 1) speech-demand)
+;(psi-rule (list (DefinedPredicate "Skip Interaction?")) (DefinedSchemaNode "Keep alive") speech-demand-satisfied (stv 1 1) speech-demand)
+
+
+(psi-rule (list (NotLink(DefinedPredicate "Skip Interaction?")) (DefinedPredicate "Interaction requested"))(DefinedSchemaNode "Interaction requested action") demand-satisfied (stv 1 1) face-demand)
+(psi-rule (list (NotLink(DefinedPredicate "Skip Interaction?")) (DefinedPredicate "New arrival sequence")) (DefinedSchemaNode "New arrival sequence action") demand-satisfied (stv 1 1) face-demand)
```

Comment:
Please limit lines to be 80- characters or less. This scrolls off the side of my web-browser, and I cannot make the browser window wide enough to read what you wrote here.